app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -1,32 +1,70 @@
 package com.lightricks.feedexercise.ui.feed
 
+import android.annotation.SuppressLint
+import android.content.Context
+import android.util.Log
 import androidx.lifecycle.*
 import com.lightricks.feedexercise.data.FeedItem
+import com.lightricks.feedexercise.data.FeedRepository
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.network.FeedApiService
 import com.lightricks.feedexercise.util.Event
-import java.lang.IllegalArgumentException
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
 
 /**
  * This view model manages the data for [FeedFragment].
  */
-open class FeedViewModel : ViewModel() {
+open class FeedViewModel(context: Context) : ViewModel() {
+    private var disposableRequest: Disposable? = null
+    private val repository: FeedRepository =
+        FeedRepository.createIfAbsent { FeedDatabase.get(context) to FeedApiService.get() }
+
     private val isLoading = MutableLiveData<Boolean>()
     private val isEmpty = MutableLiveData<Boolean>()
     private val feedItems = MediatorLiveData<List<FeedItem>>()
     private val networkErrorEvent = MutableLiveData<Event<String>>()
 
+    private val mediatorLiveData = MediatorLiveData<List<FeedItem>>()
+
     fun getIsLoading(): LiveData<Boolean> = isLoading
     fun getIsEmpty(): LiveData<Boolean> = isEmpty
     fun getFeedItems(): LiveData<List<FeedItem>> = feedItems
     fun getNetworkErrorEvent(): LiveData<Event<String>> = networkErrorEvent
 
     init {
+        feedItems.addSource(repository.feedItems) { feedItems.postValue(it) }
```

Comment:
My point was that since `feedItems` is already a `MediatorLiveData`, you could just extend the handler:
```
feedItems.addSource(feedRepository.feedItems) {
    feedItems.value = it
    isEmpty.value = it.isEmpty()
}
```